src/pcapi/scripts/beneficiary/delete_or_suspend_account_from_csv.py

```diff
@@ -0,0 +1,79 @@
+import csv
+import logging
+from typing import Optional
+
+from pcapi.core.offerers.models import UserOfferer
+from pcapi.core.users import constants
+from pcapi.core.users.api import suspend_account
+from pcapi.core.users.models import Favorite
+from pcapi.core.users.models import User
+from pcapi.models import db
+from pcapi.models.beneficiary_import import BeneficiaryImport
+from pcapi.models.beneficiary_import_status import BeneficiaryImportStatus
+from pcapi.models.deposit import Deposit
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_user_id_from_csv(user_email: str) -> Optional[int]:
+    user_id = User.query.filter(User.email == user_email).with_entities(User.id).first()
+    if user_id:
+        return user_id[0]
+    return None
+
+
+def _find_users_to_suspend(user_ids: set) -> set:
+    users_to_suspend = set()
+    for user_id in user_ids:
+        has_deposit = Deposit.query.filter(Deposit.userId == user_id).first()
+        if has_deposit:
+            users_to_suspend.add(user_id)
+    return users_to_suspend
+
+
+def _delete_users_and_favorites(user_ids: set) -> None:
+    for user_id in user_ids:
+        logger.info("[DELETING USER WITH ID] %s", user_id)
+        Favorite.query.filter(Favorite.userId == user_id).delete()
+        beneficiary_imports = BeneficiaryImport.query.filter(BeneficiaryImport.beneficiaryId == user_id).all()
+        for beneficiary_import in beneficiary_imports:
+            BeneficiaryImportStatus.query.filter(
+                BeneficiaryImportStatus.beneficiaryImportId == beneficiary_import.id
+            ).delete()
+            db.session.delete(beneficiary_import)
```

Comment:
(perfs)

N'y aurait-il pas moyen de faire moins de requêtes SQL ? Quelque chose dans le genre :

```suggestion
        BeneficiaryImportStatus.query.join(BeneficiaryImport).filter(BeneficiaryImport.beneficiaryId == user_id).delete(synchronize_session=False)
        BeneficiaryImport.query.filter_by(beneficiaryId=user_id).delete(synchronize_session=False)
```

Reply:
Impossible d'appeler delete quand join a été appelé :/
J'avais fait ça en effet

Reply:
Ach! Et comme ça :

```python

BeneficiaryImportStatus.query
    .filter(
        BeneficiaryImportStatus.beneficiaryImportId == BeneficiaryImport.id,
        BeneficiaryImport.beneficiaryId == user_id,
    )
    .delete(synchronize_session=False)
```

?